dashboard/app/models/census/other_curriculum_offering.rb

```diff
@@ -17,4 +17,104 @@
 #
 
 class Census::OtherCurriculumOffering < ApplicationRecord
+  belongs_to :school, required: true
+
+  validates_presence_of :curriculum_provider_name, :course
+  validates :school_year, presence: true, numericality: {greater_than_or_equal_to: 2015, less_than_or_equal_to: 2030}
+
+  SUPPORTED_PROVIDERS = %w(
+    TEALS
+  ).freeze
+
+  TEALS_COURSE_CODES = [
+    'Intro Full',
+    'Intro S2',
+    'Other',
+    'AP',
+    'AP CSP',
+    'Intro Both',
+    'Intro S1',
+    'Post AP',
+    'Python S2'
+  ].freeze
+
+  def self.school_id(provider_code, row_hash)
+    case provider_code
+    when 'TEALS'
+      row_hash['School ID']
+    else
+      raise ArgumentError.new("#{provider_code} is not supported.")
+    end
+  end
+
+  def self.get_courses(provider_code, row_hash)
+    case provider_code
+    when 'TEALS'
+      TEALS_COURSE_CODES.select {|course| course == row_hash['Course']}
+    else
+      raise ArgumentError.new("#{provider_code} is not supported.")
+    end
+  end
+
+  def self.seed_from_csv(provider_code, school_year, filename)
+    ActiveRecord::Base.transaction do
+      CSV.foreach(filename, {headers: true}) do |row|
+        row_hash = row.to_hash
+        input_school_id = school_id(provider_code, row_hash)
+        # Remove leading zero if it looks like an NCES ID (12 digits) because we imported NCES IDs into School.id
+        # without the leading zero.
+        lookup_school_id = input_school_id.length == 12 ? input_school_id.sub(/^0/, "") : input_school_id
+        courses = get_courses(provider_code, row_hash)
+        school = School.find_by(id: lookup_school_id)
+        if school && lookup_school_id
```

Comment:
Could we have a `school` without a `lookup_school_id`, given the previous line.

Reply:
Agreed.  Checking for `lookup_school_id` is redundant.  I've removed it.